### ISSUE TITLE
String Union Autocomplete Support

### ISSUE THREAD

**Issue by basarat** [microsoft/TypeScript#6705]
Something along: 

``` ts
type Foo = "aReallyLong" | "bReallyLong";
let foo:Foo;
foo = "aReally // Here 
```

Similar to https://github.com/Microsoft/TypeScript/issues/5602 :rose:  (also requested here https://github.com/Microsoft/TypeScript/issues/3192#issuecomment-176449974)

**Comment by olmobrutall** [comment]
:+1: 
...not sure it the autocomplete should be triggered on `=` or `"` however.

**Comment by mhegazy** [comment]
+1. PRs welcomed

**Comment by Ciantic** [comment]
Is equality autocompletion possiblity too? E.g. ` foo === "aReally` here.

**Comment by basarat** [comment]
Its been supported for a while: 

![image](https://user-images.githubusercontent.com/874898/74894268-9114df80-53e2-11ea-9601-9f8ae2cde865.png)

Thanks TS Folks :rose: :heart:
